From aa658308c90ce1a2c6c41977ce7d861ad726a25b Mon Sep 17 00:00:00 2001
From: Dave Airlie <airlied@redhat.com>
Date: Thu, 26 Nov 2009 08:55:18 +1000
Patch-mainline: 2.6.33
References: fate#310916
Git-commit: 5fbfce7fc906c4a9e3d5e0872e5d6affaca54761
Subject: [PATCH 0132/2588] drm/radeon/kms: make displayport work by
 reorganising vsemph setup.

This fix reorganises the initial DP link training slightly, and
actually makes DP work under kms here.

Signed-off-by: Dave Airlie <airlied@redhat.com>
(cherry picked from commit 5fbfce7fc906c4a9e3d5e0872e5d6affaca54761)

Signed-off-by: Takashi Iwai <tiwai@suse.de>
---
 drivers/gpu/drm/radeon/atombios_dp.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/drivers/gpu/drm/radeon/atombios_dp.c b/drivers/gpu/drm/radeon/atombios_dp.c
index 65c8239..28741d4 100644
--- a/drivers/gpu/drm/radeon/atombios_dp.c
+++ b/drivers/gpu/drm/radeon/atombios_dp.c
@@ -627,11 +627,12 @@ void dp_link_train(struct drm_encoder *encoder,
 
 	/* set initial vs/emph */
 	memset(train_set, 0, 4);
-	dp_update_dpvs_emph(radeon_connector, encoder, train_set);
 	udelay(400);
 	/* set training pattern 1 on the sink */
 	dp_set_training(radeon_connector, DP_TRAINING_PATTERN_1);
 
+	dp_update_dpvs_emph(radeon_connector, encoder, train_set);
+
 	/* clock recovery loop */
 	clock_recovery = false;
 	tries = 0;
-- 
1.7.6

